include/session/onionreq/builder.hpp

```diff
@@ -3,10 +3,46 @@
 #include <string>
 #include <string_view>
 
+#include "../network_service_node.hpp"
 #include "key_types.hpp"
 
 namespace session::onionreq {
 
+struct SnodeDestination {
+    session::network::service_node node;
+    std::optional<std::vector<session::network::service_node>> swarm;
```

Comment:
Now that the snode cache is internal this has been changed to a `std::optional<std::string> swarm_pubkey` (which is mostly needed because you can send a request directly to a node unrelated to a swarm - if `swarm_pubkey` is assigned and we get a `421` error then the cached swarm info will be updated based on the error handling logic)